backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -173,21 +213,40 @@ def delete_metadata_form_field(uri, fieldUri):
     @TenantPolicyService.has_tenant_permission(MANAGE_METADATA_FORMS)
     @MetadataFormAccessService.can_perform('UPDATE FIELDS')
     def batch_metadata_form_field_update(uri, data):
+        to_delete = []
+        to_update = []
+        to_create = []
+
+        # validate all inputs first
+        # if even one input is invalid -- decline whole batch
         for item in data:
             if item.get('metadataFormUri') != uri:
                 raise Exception('property metadataFormUri does not match form uri')
-            if 'uri' not in item:
-                MetadataFormService.create_metadata_form_field(uri=uri, data=item)
-            elif item.get('uri') is not None:
-                if item.get('deleted', False):
-                    MetadataFormService.delete_metadata_form_field(uri=uri, fieldUri=item['uri'])
-                else:
-                    MetadataFormService.update_metadata_form_field(uri=uri, fieldUri=item['uri'], data=item)
-        return MetadataFormService.get_metadata_form_fields(uri=uri)
+            if item.get('uri') is None:
+                MetadataFormParamValidationService.validate_create_field_params(item)
+                to_create.append(item)
+            elif not item.get('deleted', False):
+                MetadataFormParamValidationService.validate_update_field_params(uri, item)
+                to_update.append(item)
+            else:
+                to_delete.append(item['uri'])
+
+        # process sorted items
+        for item in to_delete:
+            MetadataFormService.delete_metadata_form_field(uri, item)
```

Comment:
need to pass as kwargs otherwise get error on delete `delete_metadata_form_field doesn't have parameter uri."`

Reply:
[8bbefd2](https://github.com/data-dot-all/dataall/pull/1480/commits/8bbefd22389d62a137fe26de245f7f2bbea15a7a)